package.json

```diff
@@ -26,28 +26,31 @@
     "watch:main": "node scripts/watch_main_and_preload.js",
     "watch:renderer": "webpack-dev-server --config webpack.config.renderer.js",
     "test": "npm-run-all test:* lint:*",
-    "test:app": "cross-env NODE_ENV=production npm run build && mocha -r babel-register --reporter mocha-circleci-reporter --recursive test/specs",
+    "test:app": "cross-env NODE_ENV=production npm run build && mocha -r @babel/register --reporter mocha-circleci-reporter --recursive test/specs",
     "package:all": "cross-env NODE_ENV=production npm-run-all check-build-config package:windows package:mac package:linux",
     "package:windows": "cross-env NODE_ENV=production npm-run-all check-build-config build && build --win --x64 --ia32 --config.extraMetadata.name=mattermost --publish=never",
     "package:mac": "cross-env NODE_ENV=production npm-run-all check-build-config build && build --mac --publish=never",
     "package:linux": "cross-env NODE_ENV=production npm-run-all check-build-config build && build --linux --x64 --ia32 --config.extraMetadata.name=mattermost-desktop --publish=never",
     "manipulate-windows-zip": "node scripts/manipulate_windows_zip.js",
     "lint:js": "eslint --ignore-path .gitignore --ignore-pattern node_modules --ext .js --ext .jsx .",
     "fix:js": "eslint --ignore-path .gitignore --ignore-pattern node_modules --quiet --ext .js --ext .jsx . --fix",
-    "check-build-config": "node -r babel-register scripts/check_build_config.js"
+    "check-build-config": "node -r @babel/register scripts/check_build_config.js"
   },
   "devDependencies": {
     "7zip-bin": "^4.0.2",
-    "@storybook/react": "^3.4.6",
-    "babel-core": "^6.26.3",
+    "@babel/core": "^7.1.6",
+    "@babel/plugin-proposal-class-properties": "^7.1.0",
+    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
+    "@babel/preset-env": "^7.1.6",
+    "@babel/preset-react": "^7.0.0",
+    "@babel/register": "^7.0.0",
+    "@storybook/addon-actions": "^4.0.9",
+    "@storybook/react": "^4.0.9",
     "babel-eslint": "^8.2.3",
-    "babel-loader": "^7.1.4",
-    "babel-plugin-transform-object-rest-spread": "^6.26.0",
-    "babel-preset-env": "^1.7.0",
-    "babel-preset-react": "^6.24.1",
+    "babel-loader": "^8.0.4",
     "chai": "^4.1.2",
     "cross-env": "^5.1.6",
-    "css-loader": "^0.28.11",
+    "css-loader": "^1.0.1",
     "devtron": "^1.4.0",
     "electron": "2.0.12",
```

Comment:
Please make sure to use `^` for electron, otherwise you are asking for a particular version of electron and I cannot build against a system one.

Reply:
Well, I think a particular version will be eventually used by yarn.lock or package-lock.json anyway. Is that okay? I'm not sure "I cannot build against a system one." though, would you describe in detail?

This PR focuses on build tools stack only, so I will update #892 or make another PR.

Reply:
Actually if you are asking to use a specific version of electron, this means on GNU/Linux distributions, I need to builtin that specific electron version to make the app work and avoid version mismatches.
While at the contrary, if you use ^2.0.12, the latest 2.0.14 from the 2.x branch will be pulled (at this time 2.0.14) and I will be able to use version 2.0.14 from my system and depend on it rather than build it in the final application distribution package :)

Reply:
I think that means that untested version of electron will be used in the final packages though, is that acceptable? In past versions of electron and electron-builder, I had met that kind of trouble (in recent versions, actually it's a rare case).

Reply:
It's acceptable because with the new build system , I plan to run CI builds every day. So these "potential untested" electron version will be something that will be taken care of.

And of the record, I don't think the `^` operator takes betas version into account, only full/real release, but this needs to be checked out. Any idea about this topic?

Reply:
@yuya-oc Indeed. Then no more objection. We can merge! :D